src/components/Footer/Footer.js

```diff
@@ -28,8 +36,19 @@ const Footer = ({
   blockStatus,
   loadingVote,
   showScoreButton = false,
-  onPressScore
+  onPressScore,
+  isShowDM = false
 }) => {
+  const {sendMessageDM} = useDMMessage();
+  const [profile] = React.useContext(Context).profile;
+
+  const [loading, setLoading] = React.useState({
+    loadingDm: false,
+    loadingDmAnon: false
+  });
+  const refSheet = React.useRef();
+  const {createSignChat} = useCreateChat();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces state management for loading states and hooks for direct messaging and chat creation. It also adds a ref for the BottomSheetMenu component. These changes seem to be well implemented and follow best practices for using React hooks and refs.


<!-- This is an auto-generated comment by OSS CodeRabbit -->